cms/djangoapps/contentstore/views/tests/test_course_index.py

```diff
@@ -329,6 +330,136 @@ def test_course_outline_with_display_course_number_as_none(self):
+        self.staff, self.staff_password = self.create_non_staff_user()
+        for course in (self.course, self.active_course, self.archived_course):
+            CourseStaffRole(course.id).add_users(self.staff)
+
+    def check_index_page_with_query_count(self, separate_archived_courses, org, mongo_queries, sql_queries):
+        """
+        Checks the index page, and ensures the number of database queries is as expected.
+        """
+        with self.assertNumQueries(sql_queries):
+            with check_mongo_calls(mongo_queries):
+                self.check_index_page(separate_archived_courses=separate_archived_courses, org=org)
+
+    def check_index_page(self, separate_archived_courses, org):
+        """
+        Ensure that the index page displays the archived courses as expected.
+        """
+        index_url = '/home/'
+        index_params = {}
+        if org is not None:
+            index_params['org'] = org
+        index_response = self.client.get(index_url, index_params, HTTP_ACCEPT='text/html')
+        self.assertEquals(index_response.status_code, 200)
+
+        parsed_html = lxml.html.fromstring(index_response.content)
+        course_tab = parsed_html.find_class('courses')
+        self.assertEqual(len(course_tab), 1)
+        course_links = course_tab[0].find_class('course-link')
+        course_titles = course_tab[0].find_class('course-title')
+        archived_course_tab = parsed_html.find_class('archived-courses')
+
+        if separate_archived_courses:
+            # Archived courses should be separated from the main course list
+            self.assertEqual(len(archived_course_tab), 1)
+            archived_course_links = archived_course_tab[0].find_class('course-link')
+            archived_course_titles = archived_course_tab[0].find_class('course-title')
+            self.assertEqual(len(archived_course_links), 1)
+            self.assertEqual(len(archived_course_titles), 1)
+            self.assertEqual(archived_course_titles[0].text, 'Archived Course')
+
+            self.assertEqual(len(course_links), 2)
+            self.assertEqual(len(course_titles), 2)
+            self.assertEqual(course_titles[0].text, 'Active Course 1')
+            self.assertEqual(course_titles[1].text, 'Active Course 2')
+        else:
+            # Archived courses should be included in the main course list
+            self.assertEqual(len(archived_course_tab), 0)
+            self.assertEqual(len(course_links), 3)
+            self.assertEqual(len(course_titles), 3)
+            self.assertEqual(course_titles[0].text, 'Active Course 1')
+            self.assertEqual(course_titles[1].text, 'Active Course 2')
+            self.assertEqual(course_titles[2].text, 'Archived Course')
+
+    @ddt.data(
+        # Staff user has course staff access
+        (True, 'staff', None, 4, 20),
+        (False, 'staff', None, 4, 20),
+        # Base user has global staff access
+        (True, 'user', ORG, 3, 16),
```

Comment:
Yes, they changed when I rebased, so these tests might be too brittle to keep?

What I wanted was a way to find out how many queries were required with the feature disabled, and then ensure that enabling the feature didn't cause any more queries to happen.  But I couldn't figure out how to do that with the existing helpers?